### PR TITLE
RUBY-1311 Test kerberos in evergreen

### PR DESCRIPTION
**Pull request by saghm**

### CODE REVIEW

**Comment by p-mongo** [comment]
Can we pick one of "kerberos", "gssapi" and "enterprise auth" and use that everywhere?

**Comment by saghm** [comment]
I've eliminated usage of "GSSAPI" in favor of kerberos.

The reason for the distinction between "enterprise auth" and "kerberos" is that we'll likely add LDAP tests at some point, and we'll also want to run those in the enterprise auth variant/task. I've renamed a couple of things in the evergreen config and the spec repo to try to clear up the distinction between them.

File: spec/enterprise_auth/kerberos_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  let(:user) do
+   "#{ENV['SASL_USER']}%40#{ENV['SASL_HOST'].upcase}"
+  end
+
+  let(:host) do
+    "#{ENV['SASL_HOST']}"
+  end
+
+  let(:gssapi_db) do
+    "#{ENV['GSSAPI_DB']}"
```
Comment (p-mongo) [inline spec/enterprise_auth/kerberos_spec.rb]: What about this env var?

**Comment by saghm** [comment]
Currently, the names of the environmental variables are just copied from the names of the Evergreen project variables, and I think it makes sense to keep them the same as each other. I guess we can change those names in the Evergreen project if you'd prefer that.

**Comment by saghm** [comment]
Sounds good; I'll change our evergreen config variables to use "kerberos" instead of "gssapi" and then update the code accordingly

**Comment by saghm** [comment]
Everything good to merge now?

**Comment by p-mongo** [comment]
:+1:

**Comment by saghm** [comment]
Awesome, thanks!

**Pull request merged**